cli/lib/kontena/cli/stacks/install_command.rb

```diff
@@ -46,18 +50,25 @@ def install_dependencies
       dependencies.each do |dependency|
         target_name = "#{stack_name}-#{dependency['name']}"
         caret "Installing dependency #{pastel.cyan(dependency['stack'])} as #{pastel.cyan(target_name)}"
-        cmd = ['stack', 'install', '-n', target_name, '--parent-name', stack_name]
+        cmd = ['stack', 'install', '-n', target_name, '--parent-name', stack_name, '--no-deploy']
 
         dependency['variables'].merge(dependency_values_from_options(dependency['name'])).each do |key, value|
           cmd.concat ['-v', "#{key}=#{value}"]
         end
 
-        cmd << '--no-deploy' unless deploy?
-
         cmd << dependency['stack']
         Kontena.run!(cmd)
       end
+    end
+
+    def deploy_dependencies
+      dependencies = loader.dependencies
+      return if dependencies.nil?
```

Comment:
If loader returned [] instead of nil, these could be just `loader.dependencies.each`

Reply:
Yes, but I'm not sure what side-effects that would cause.